<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Footnotes

In [7]:
while True:
    print('*', end='')
#interrupt kernel

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

In [8]:
print = 'PRINT'

In [9]:
print('Hello World')

TypeError: 'str' object is not callable

In [ ]:
#now i restartkernal

In [1]:
print('Hello World')

Hello World


In [4]:
squares = [i**2 for i in range(10)]
print (squares)


[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [8]:
squares = list()
for i in range(10):
    squares.append(i**2)

print (squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [5]:
%%timeit
squares = [i**2 for i in range(10)]

1.22 μs ± 128 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [6]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

1.36 μs ± 132 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


1.36 μs ± 132 ns - mean time it takes to execute the entire code with standard deviation 
per loop- it runs the code multiple times and each time(loop) it takes this long
(mean ± std. dev. of 7 runs - the functions runs the code from the start for 7 times to ensure no funky business
, 1,000,000 loops each) - each time it runs the code from the start it repeats for this many times

